io-sim/src/Control/Monad/IOSim/Types.hs

```diff
@@ -793,10 +781,19 @@ data SimEventType
                        (Maybe Effect)    -- effect performed (only for `IOSimPOR`)
   | EventTxWakeup      [Labelled TVarId] -- changed vars causing retry
 
-  | EventTimerCreated   TimeoutId TVarId Time
-  | EventTimerUpdated   TimeoutId        Time
-  | EventTimerCancelled TimeoutId
-  | EventTimerExpired   TimeoutId
+  | EventThreadDelay        Time
+  | EventThreadDelayExpired
```

Comment:
Shouldn't we also have `EventThreadDelayFired`?

Reply:
`EventThreadDelayExpired` is supposed to denote `EventThreadDelayFired`

Reply:
ok, so it's better to conform to the same naming convention and use `EventThreadDelayFired`.